build-support/bin/native/bootstrap.sh

```diff
@@ -19,7 +19,10 @@ readonly NATIVE_ENGINE_VERSION_RESOURCE="${REPO_ROOT}/src/python/pants/engine/su
 
 # N.B. Set $MODE to "debug" to generate a binary with debugging symbols.
 readonly MODE="release"
```

Comment:
These 2 things would be good:
+ `readonly MODE="${MODE:-release}"` so this can be done via `MODE=debug ./pants ...`.
+ mix the MODE into the hashing so `MODE=debug ./pants ...` actually re-builds when there has been a prior `--release` build.